apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx

```diff
@@ -378,33 +379,69 @@ const PoolCandidatesTable = ({
     ?.filter(notEmpty)
     .map((skill) => skill.id);
 
-  const querySelected = async (action: SelectingFor) => {
-    setSelectingFor(action);
-    setIsSelecting(true);
-    return client
-      .query(PoolCandidatesTable_SelectPoolCandidatesQuery, {
-        ids: selectedRows,
-      })
-      .toPromise()
-      .then((result) => {
-        const poolCandidates: PoolCandidate[] = unpackMaybes(
-          result.data?.poolCandidates,
+  const isPoolCandidate = (
+    candidate: Error | PoolCandidate | null,
+  ): candidate is PoolCandidate =>
+    candidate !== null && !(candidate instanceof Error);
+
+  const batchLoader = new DataLoader<string, PoolCandidate | null>(
+    async (ids) => {
+      const batchSize = 8;
+      const batches = [];
+
+      for (let i = 0; i < ids.length; i += batchSize) {
+        const batchIds = ids.slice(i, i + batchSize);
+        batches.push(
+          client
+            .query<{
+              poolCandidates: PoolCandidate[];
+            }>(PoolCandidatesTable_SelectPoolCandidatesQuery, {
+              ids: batchIds,
+            })
+            .then((result) => result.data?.poolCandidates ?? []),
         );
+      }
 
-        if (result.error) {
-          toast.error(intl.formatMessage(errorMessages.unknown));
-        } else if (!poolCandidates.length) {
-          toast.error(intl.formatMessage(adminMessages.noRowsSelected));
-        }
+      try {
+        const batchResults = await Promise.all(batches);
+        const candidates = batchResults.flat(); // Flatten the array
 
-        setSelectedCandidates(poolCandidates);
-        setIsSelecting(false);
-        setSelectingFor(null);
-        return poolCandidates;
-      })
-      .catch(() => {
-        toast.error(intl.formatMessage(errorMessages.unknown));
-      });
+        return ids.map(
+          (id) => candidates.find((candidate) => candidate.id === id) ?? null,
+        );
+      } catch (error) {
+        return ids.map(() => null); // Return null for all IDs in case of an error
```

Comment:
This will help complete the query in case of any error and download the profiles for successful batches.
I believe this is a better idea instead of giving an empty excel sheet on the error occurrence. 
Idk if I can still loop thru this failure batch and mine that candidates with profile  and return null only for errored candidates. 🤔